fish_prompt.fish

```diff
@@ -238,23 +238,32 @@ end
 
 function __bobthefish_prompt_vagrant -S -d 'Display Vagrant status'
   [ "$theme_display_vagrant" = 'yes' -a -f Vagrantfile ]; or return
-  if type -q VBoxManage
-    __bobthefish_prompt_vagrant_vbox
-  else if grep vmware_fusion Vagrantfile >/dev/null ^&1
-    __bobthefish_prompt_vagrant_vmware
+  set provider (__bobthefish_vagrant_provider)
+  switch "$provider"
+    case 'virtualbox'
+      __bobthefish_prompt_vagrant_vbox
+    case 'vmware_fusion'
+      __bobthefish_prompt_vagrant_vmware
+    case 'parallels'
+      __bobthefish_prompt_vagrant_parallels
   end
 end
 
-function __bobthefish_vagrant_ids -S -d 'List Vagrant machine ids'
-  for file in .vagrant/machines/**/id
+function __bobthefish_vagrant_provider -S -d 'Get (first) Vagrant provider'
```

Comment:
To be honest, I only use one Vagrant provider, so I've never tested how it works when you have multiple. Is it possible for there to be VMs from two or more Vagrant providers in the same directory? If so, is there any reason you'd care about all of them?